data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessor.java

```diff
@@ -119,21 +160,65 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
         event.getMetadata().addTags(tagsOnFailure);
       }
     }
+    geoIPDatabaseReader.close();
```

Comment:
This could be missed if you have an exception that isn't caught. I recommend you use try-with-resources.

```
try(final GeoIPDatabaseReader geoIPDatabaseReader = geoIPProcessorService.getGeoIPDatabaseReader()) {
 ...
}
```

The try block will call `.close()`, so you won't even need this line.

Reply:
I will address this in a follow-on PR.